docs/source/deploypeer/peerplan.md

```diff
@@ -0,0 +1,94 @@
+
+This approach limited chaincode implementations to a handful of languages, required Docker to be part of the deployment environment, prevented running chaincode as a long-running server process, and required that the peer have privileged access to the chaincode container.
+
+Starting with Fabric 2.0, External Builders and Launchers enable operators to extend the peer with programs that can build, launch, and discover chaincode. To leverage this capability on peers that already exist you will need to create your own buildpack and then modify `core.yaml` to include a new externalBuilder configuration element which lets the peer know an external builder is available.
+
+## Gossip
+
+Peers leverage the [gossip data dissemination protocol](../gossip.html) to broadcast ledger and channel data in a scalable fashion. Gossip messaging is continuous, and each peer on a channel is constantly receiving data from multiple peers, including peers in other organizations (if cross-organization gossip is enabled).
+
+For peer gossip to work you need to configure four parameters. Three of them --- `peer.gossip.bootstrap`, `peer.gossip.endpoint`, `peer.gossip.externalEndpoint` --- are in the peer’s `core.yaml` file. The fourth enables gossip between organization by specifying an anchor peer in the channel configuration.
+
+To reduce network traffic, Fabric v2.2 includes the recommendation that peers get their blocks from the ordering service instead of through gossip dissemination among peers (with the exception of private data transactions, which are still sent from peer to peer). To get all non-private data blocks from the orderer, you must configure the following three parameters in the `core.yaml` file:
+
+* `useLeaderElection = false`
+* `orgLeader = true` for ( you must designate one peer in the org to be the leader)
+* `peer.gossip.state.enabled = false`
+
+If you have only one peer in your organization configured with `orgLeader=true`, then that peer will get blocks from ordering service and distribute them to all the other peers in the org using gossip. If all peers have `orgLeader=true` (recommended), then each peer will get blocks from the ordering service.
+
+### Service Discovery
+
+In any network it is possible that peer nodes can be down for maintenance, unreachable due to network issues, or the peer ledger has fallen behind while being offline. For this reason, Fabric includes a “discovery service” that enables client applications that use the SDK to locate good candidate peers to target with endorsement requests. If service discovery is not enabled, if a client application targets a peer that is offline, the request fails and will need to be resubmitted to another peer. The discovery service runs on peers and uses the network metadata information maintained by the gossip communication layer to find out which peers are online and can be targeted for requests.
```

Comment:
If service discovery is not enabled, when a client application targets a peer that is offline,